[MyMeta:web] OpenGraph meta tags for embeds from links

Vyvy-vi opened the pull request:
## Related issue
Closes #600

## What is the scope of this PR?
This PR tweaks the metagame website(`TheGame/packages/web`)
(touches the `pages` folder in there)

## What does this PR propose?
MetaGame is such a cool project. Perhaps we should make our links posted in chat look cool too. 🙃 

The meta tags and OpenGraph metadata added to the pages, help discord and other platforms render neat cards/embeds for the links.

## Screenshots
![Embeds generated for some of the routes](https://user-images.githubusercontent.com/62864373/119745833-9cd83700-beac-11eb-8bcb-2ee258e8b9e0.png)
![Embeds for main site](https://user-images.githubusercontent.com/62864373/119745849-a6619f00-beac-11eb-8f5d-7415a9cb016f.png)

hammadj commented on packages/web/pages/guilds.tsx:
```diff
@@ -18,6 +19,19 @@ export const getStaticProps = async () => {
 
 const GuildsPage: React.FC<Props> = ({ guilds }) => (
   <PageContainer>
+    <Head>
+      <meta name="title" content="Metagame's Guilds" />
+      <meta
+        name="description"
+        content="A Massive Online Coordination Game! Guilds participating in Metagame."
+      />
+      <meta property="og:title" content="Metagame's Guilds" />
+      <meta
+        property="og:description"
+        content="A Massive Online Coordination Game! Guilds participating in Metagame."
+      />
+      <meta property="og:url" content="https://my.metagame.wtf/guilds" />
+    </Head>
```
Would be nice to turn this into a component with props for title / description etc that have defaults. Will ensure consistency between pages that they all use the same tags

Vyvy-vi replied:
I'll look into making components :+1:

Do we need to pull in some plugin like [next-seo](https://github.com/garmeeh/next-seo)?

hammadj replied:
> I'll look into making components 👍
> 
> Do we need to pull in some plugin like [next-seo](https://github.com/garmeeh/next-seo)?

Not necessary but that might make it easier to implement, feel free to use it

Vyvy-vi replied:
I created a component, in [`components/Seo.tsx`](https://github.com/Vyvy-vi/TheGame/blob/feat/og-meta-tags/packages/web/components/Seo.tsx) - commit - 7ebb213

Is the implementation alright?

hammadj reviewed (commented):
Looks good, I would just refactor to make a component that takes props and is reused everywhere instead of duplicating the tags on each page, will make it easier to maintain / keep consistent

Vyvy-vi commented:
![Screenshot 2021-06-04 at 4 06 28 AM](https://user-images.githubusercontent.com/62864373/120720777-4ab99600-c4ea-11eb-8ba6-69225e806ed1.png)

dan13ram commented:
@Vyvy-vi Looks like there is conflict. Anyway I've approved the vercel deployment for now.

Request you to ping in discord for faster response, Cheers!

dan13ram commented:
Now that you're a member of the org, for future PRs you can work in branches in the same repo so that previews don't need to be approved.
Also it helps reviewing in case we need to pull changes locally for testing :)

dan13ram commented:
Vercel deployment failed with the following logs for every player: 
```
20:06:13.603 | @metafam/web: Error occurred prerendering page "/player/alec". Read more: https://nextjs.org/docs/messages/prerender-error
-- | --
20:06:13.604 | @metafam/web: TypeError: Cannot read property 'replace' of null
20:06:13.604 | @metafam/web:     at PlayerPage (/vercel/path0/packages/web/.next/server/pages/player/[username].js:4108:49)
20:06:13.604 | @metafam/web:     at d (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:498)
20:06:13.604 | @metafam/web:     at $a (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:39:16)
20:06:13.604 | @metafam/web:     at a.b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:44:476)
20:06:13.604 | @metafam/web:     at a.b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:44:18)
20:06:13.608 | @metafam/web:     at renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:54:364)
20:06:13.608 | @metafam/web:     at Object.renderPage (/vercel/path0/node_modules/next/dist/next-server/server/render.js:54:854)
20:06:13.608 | @metafam/web:     at Function.getInitialProps (/vercel/path0/packages/web/.next/server/pages/_document.js:325:19)
20:06:13.608 | @metafam/web:     at Function.getInitialProps (/vercel/path0/packages/web/.next/server/pages/_document.js:906:85)
20:06:13.608 | @metafam/web:     at loadGetInitialProps (/vercel/path0/node_modules/next/dist/next-server/lib/utils.js:5:101)
```

Could you rebase and then run `yarn web:build` locally to see if the build works properly :)

dan13ram commented on packages/web/pages/patrons.tsx:
```diff
@@ -18,6 +19,11 @@ export const getStaticProps = async () => {
   <PageContainer>
+    <HeadComponent
+      title="Metagame's Patrons"
+      description="Metagame is a Massive Online Coordination ame! Metagame's Patrons enable us to succeed by helping us with funds."
```
```suggestion
      description="Metagame is a Massive Online Coordination Game! Metagame's Patrons enable us to succeed by helping us with funds."
```

Vyvy-vi replied:
resolved by 787998b

dan13ram commented on packages/web/pages/player/[username].tsx:
```diff
@@ -100,6 +101,15 @@ const PlayerPage: React.FC<Props> = ({ player }) => {
     <PageContainer>
+      <HeadComponent
+        title={`Metagame profile for ${player.username}`}
+        description={`${player.box_profile?.description} || ''`.replace(
```
```suggestion
        description={(player.box_profile?.description || '').replace(
```

dan13ram replied:
There are existing methods you can instead use here 
`getPlayerDescription` from `https://github.com/MetaFam/TheGame/blob/develop/packages/web/utils/playerHelpers.ts`

Event in the title instead of username you could use `getPlayerName`

Vyvy-vi replied:
for some reason, when I add this suggestion, the automated tests in the pre-commit hook fail with -
```
@metafam/web: /Users/vyom/TheGame/packages/web/pages/player/[username].tsx
@metafam/web:   106:53  error  Parsing error: ')' expected
@metafam/web: ✖ 1 problem (1 error, 0 warnings)
```

Vyvy-vi replied:
Though, I think using the helpers would be much cleaner :+1:

Vyvy-vi replied:
d9a4e2a

dan13ram commented on packages/web/pages/player/[username].tsx:
```diff
@@ -100,6 +101,15 @@ const PlayerPage: React.FC<Props> = ({ player }) => {
+          ' ',
+        )}
+        url={`https://my.metagame.wtf/player/${player.username}`}
+        img={player.box_profile?.imageUrl || ''}
```
Here also you can use the `getPlayerImage` helper

Vyvy-vi replied:
resolved by d9a4e2a

dan13ram commented on packages/web/pages/quests.tsx:
```diff
@@ -51,6 +52,11 @@ const QuestsPage: React.FC<Props> = () => {
     <PageContainer>
+      <HeadComponent
+        title="Metagame's Quests"
+        description="Metagame is a massive online coordination game. Metagame has some epic quests going on!"
```
Could we keep the first sentence of the description consistent across all pages? 
I see these three different occurrences which can all be replaced with one of them:

1. A Massive Online Coordination Game!
2. Metagame is a Massive Online Coordination Game!
3. Metagame is a massive online coordination game.

Vyvy-vi replied:
resolved by 787998b

dan13ram reviewed (changes requested):
Looking good, requested some minor changes.

dan13ram commented:
Hey great work. Looks like a rebase is necessary though

Vyvy-vi commented:
wupps, I should have squashed the commits after rebasing

dan13ram commented:
Hey looks like there are still some conflicts. Have to rebased to `develop`?

Vyvy-vi commented:
> Hey looks like there are still some conflicts. Have you rebased to `develop`?

yes I rebased to develop

dan13ram commented:
![Screenshot from 2021-06-11 13:57:59](https://user-images.githubusercontent.com/10012209/121656414-154e1300-cabd-11eb-86d2-fe1cdf5cddbe.png)

@Vyvy-vi  The conflicts would show up only when `rebase and merge` is selected.
Please rebase with the latest `develop` once again.

Vyvy-vi commented:
Maybe you can Squash and merge?
(The conflicts that would arise from re-basing and merging would need to be addressed during the rebase, even if I have the latest develop rebased on the branch(which I do have))

Vyvy-vi commented:
> @Vyvy-vi please check again, because I just pulled your code locally and its not rebased to the latest `develop` branch.

![Screenshot 2021-06-11 at 3 41 00 PM](https://user-images.githubusercontent.com/62864373/121670792-754bb600-cacb-11eb-901e-0af57877e6e8.png)

^Latest commits from `develop` branch, that are in my git log on `feat/og-meta-tags` branch

Thus, it is most probably rebased. (If you rebase it again via the github UI, of course there will be merge conflicts since both branches changed the same files, and my branch is supposed to over-write the changes on the `develop` branch. I already rebased it to the latest version(if it wasn't rebased, the github UI would show - `This branch has conflicts that must be resolved`))

You could choose `Squash and Merge`, and there won't be much of an issue.
(since I already rebased, and fixed the conflicts, this would just add one squashed merge commit. Perhaps, you can edit the commit message to something like `feat: add Meta tags`)
